toolchain/semantics/semantics_handle_operator.cpp

```diff
@@ -51,7 +51,14 @@ auto SemanticsHandleInfixOperator(SemanticsContext& context,
               resume_block_id));
       return true;
     }
-
+    case TokenKind::Equal: {
+      // TODO: handle complex assignment expression such as `a = a + 1`.
+      context.ImplicitAsRequired(
+          parse_node, lhs_id, context.semantics_ir().GetNode(rhs_id).type_id());
```

Comment:
I think we should do this the other way around: convert the RHS to the type of the LHS.